### PR TITLE
Async rework

### PR DESCRIPTION
**Pull request by weliasz**

### CODE REVIEW

File: splunk_connect_for_snmp_traps/manager/mib_server_client.py [hunk 1]
```diff
@@ -46,39 +43,26 @@ def get_translation(var_binds, mib_server_url):
...
-        resp = session.post(TRANSLATION_URL, headers=headers, data=payload)
-        # If varBinds gets translated, overide it with the translated one
-        if resp.status_code == 200:
-            trap_event_string = resp.text
-        if resp.status_code != 200:
-            logger.error(f"[-] Mib Server API Error with code: {resp.status_code}")
+        session = ClientSession()
+        resp = await session.post(TRANSLATION_URL, headers=headers, data=payload)
+        if resp.status == 200:
+            trap_event_string = await resp.text()
```
Comment (okuzhel-splunk) [inline splunk_connect_for_snmp_traps/manager/mib_server_client.py:59]: why do we need await on resp.text()? Does it make some additional call or is that just a value returned from resp?

File: splunk_connect_for_snmp_traps/manager/trap_server.py [hunk 2]
```diff
@@ -191,24 +199,24 @@ def snmp_callback_function(
...
             pass
 
         header["Agent_Address"] = device_ip
 
-        # Send API call to SNMP MIB server to get var_binds translated
-        # mib_server_url = self._server_config["snmp"]["mib-server"]["url"]
-        mib_server_url = os.environ["MIBS_SERVER_URL"]
-        trap_event_string = get_translation(var_binds, mib_server_url)
+        loop = asyncio.get_event_loop()
+        tasks = get_translation(var_binds, os.environ["MIBS_SERVER_URL"])
```
Comment (okuzhel-splunk) [inline splunk_connect_for_snmp_traps/manager/trap_server.py:217]: get_translation is async and we are in a sync part of code.  So calling it returns some kind of hook that we are waiting for in asyncio.gather, right?

File: splunk_connect_for_snmp_traps/manager/mib_server_client.py (hunk 1 above)
Comment (weliasz) [reply splunk_connect_for_snmp_traps/manager/mib_server_client.py:59]: text is an async method. I spent a few hours debugging to find out that this is not a field. And as this is async method it need to be awaited.

File: splunk_connect_for_snmp_traps/manager/trap_server.py (hunk 2 above)
Comment (weliasz) [reply splunk_connect_for_snmp_traps/manager/trap_server.py:217]: From docs:
run_until_complete """Run the event loop until a Future is done.

gather: """Return a future aggregating results from the given coroutines/futures.

Maybe gather is not necessary, I got it from some code examples and there it was used for multiple coroutines and here we have just one so maybe loop.run_until_complete(tasks) would be enough

File: splunk_connect_for_snmp_traps/manager/mib_server_client.py [hunk 3]
```diff
@@ -46,36 +43,22 @@ def get_translation(var_binds, mib_server_url):
...
-        session = requests.Session()
-        session.mount("https://", adapter)
-        session.mount("http://", adapter)
-        resp = session.post(TRANSLATION_URL, headers=headers, data=payload)
-        # If varBinds gets translated, overide it with the translated one
-        if resp.status_code == 200:
-            trap_event_string = resp.text
-        if resp.status_code != 200:
-            logger.error(f"[-] Mib Server API Error with code: {resp.status_code}")
+        async with ClientSession() as session:
```
Comment (okuzhel-splunk) [inline splunk_connect_for_snmp_traps/manager/mib_server_client.py:56]: we did change a logic here a little removing retries completely. 
Was that intended or not? If not we can add them back with https://github.com/inyutin/aiohttp_retry

Comment (weliasz) [reply splunk_connect_for_snmp_traps/manager/mib_server_client.py:57]: Contrary to "requests" lib aiohttp does not support retries functionality. I've also found above mentioned library but I had objections on using library created by random person on Github.

Comment (okuzhel-splunk) [reply splunk_connect_for_snmp_traps/manager/mib_server_client.py:57]: but we are definitely changing the logic here so we need to at least create a ticket to add it back

**Review by okuzhel-splunk — APPROVED** [review]

**Pull request merged**